spec/system/admin/product_import_spec.rb

```diff
@@ -675,14 +680,20 @@
       expect(page).to have_selector '.created-count', text: '2'
       expect(page).not_to have_selector '.updated-count'
 
-      puts "TODO: migrate to v3"
-      Flipper.disable(:admin_style_v3) # disabling BUU for legacy products page
+      default_variant_selector = "tr:has(input[aria-label=Name][value='Carrots'])"
+
       visit spree.admin_products_path
 
-      within "#p_#{Spree::Product.find_by(name: 'Carrots').id}" do
-        expect(page).to have_input "product_name", with: "Carrots"
-        expect(page).to have_select "variant_unit_with_scale", selected: "Weight (lb)"
-        expect(page).to have_content "5" # on_hand
+      carrots = Spree::Product.find_by(name: 'Carrots')
+
+      within "#product_#{carrots.id}" do
+        expect(page).to have_input("[products][2][variants_attributes][0][display_name]",
+                                   text: "Carrots")
+        expect(page).to have_input("[products][2][variants_attributes][][0][unit_to_display]",
+                                   text: "1 lb")
```

Comment:
Also here we can use the column labels to refer:

```suggestion
        expect(page).to have_input("Name", text: "Carrots") # (This is now already asserted by row_containing_name above)
        expect(page).to have_input("Unit", text: "1 lb")
```

Reply:
I find that the biggest challenge is to write simple code ;)

And it's a balance deciding how much time to spend on it! I think that is fine as it is.